### ISSUE TITLE
KeyError in Earthquake Raster vs Population Raster

### ISSUE THREAD

**Issue by ivanbusthomi** [inasafe/inasafe#3803]
### Problem
I got this error when running earthquake raster on population raster.
```
Problem
------------
The following problem(s) were encountered whilst running the analysis.
KeyError

Details
------------
These additional details were reported when the problem occurred.
classification

Diagnostics
------------
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1021, in run self.style()
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1678, in style self.debug_mode)
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\style.py", line 117, in generate_classified_legend hazard_classification = hazard.keywords['classification']
```

### Environment
* InaSAFE : 4.0 develop
* QGIS : 2.14.10
* OS : Windows
* Data used : West Sumatra dataset
  * Exposure : Population Raster
  * Hazard : Earthquake raster
  * Aggregation : villages

### Proposed Solution
Make it Happy
___

CC @ismailsunni @timlinux @Charlotte-Morgan @samnawi @felix-yew

**Comment by samnawi** [comment]
Hi @ivanbusthomi I also found the "Key Error" problem in Flood raster VS Population Raster and Tsunami raster VS population raster.

Hi @Gustry Should I open new ticket for it, or it's still relevant with this ticket?

**Comment by Gustry** [comment]
Can you open a new one @samnawi ?

EQ IF is a special one so your error might be different. Thanks

**Comment by Gustry** [comment]
This has been fixed. I'm going to close this one.

**Comment by Charlotte-Morgan** [comment]
hi @Gustry - i got key error for earthquake raster on pop raster with aggregation boundaries this morning

**Comment by Gustry** [comment]
Is-it the same KeyError ?
Can you copy/paste logs ?

**Comment by ivanbusthomi** [comment]
Hi @Gustry , actually I tried using west sumatra dataset in Real world data and it works. But when I tried using another data set, I got a key error. these are the diagnostics
```
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1067, in run self._run()
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 93, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1228, in _run self.damage_curve_analysis()
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 93, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1346, in damage_curve_analysis self.earthquake_raster_population_raster()
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 93, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1394, in earthquake_raster_population_raster exposed, self.aggregation, earthquake_function())
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 93, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/busthi/.qgis2/python/plugins\inasafe\safe\impact_function\earthquake.py", line 258, in make_summary_layer stats_aggregation = exposed_per_agg_zone[agg_zone]
```

**Comment by Gustry** [comment]
What is the other dataset @ivanbusthomi?

**Comment by ivanbusthomi** [comment]
Hi @Gustry this is the data:
[maluku_clip.zip](https://github.com/inasafe/inasafe/files/813026/maluku_clip.zip)

**Comment by Gustry** [comment]
So we got two kinds of KeyError in this ticket: 
* In the Aceh scenario from @Charlotte-Morgan it was due to an invalid aggregation layer. In the logs, I could see some message saying that geometries were still invalid after cleaning. I ran a `Select by expression` with `$geometry IS NULL` and I got two results in the layer: 
![screen shot 2017-03-03 at 00 10 01](https://cloud.githubusercontent.com/assets/1609292/23532793/bf2cc8f6-ffad-11e6-9998-d811d0bb2acc.png)
We can't zoom to these features, they are without geometry.

* In the Maluku scenario from @ivanbusthomi . This is because the aggregation layer is much bigger than the hazard and exposure layer. I'm gonna make a fix for this, and the same fix gonna fix the other KeyError above
